### PR TITLE
[flutter_tools] flutter logs no longer requires supported device

### LINKED ISSUE #47996: "flutter logs" claims "No supported devices connected." when not run in directory with pubspec.yaml
Steps to reproduce:
1. Plug in a device.
2. `flutter create dummy`
3. `cd dummy/lib`
4. `flutter logs`

Expected results:
Logs should start, probably not showing much unless you have a debug-mode Flutter app running already on the device.

Actual results:
```
No supported devices connected.
```

Works fine if you run it from a directory with a `pubspec.yaml`, but that command shouldn't need a Flutter package to find the devices on which logs are available.

### LINKED ISSUE #63550: "flutter logs" has confused error message
I ran `flutter logs` and got the following, which is nonsensical:

```
$ flutter logs
No supported devices connected.
The following devices were found, but are not supported by this project:
Lenovo TB 8504F (mobile) • HGAFGRP2   • android-arm64  • Android 7.1.1 (API 25)
Web Server (web)         • web-server • web-javascript • Flutter Tools
Chrome (web)             • chrome     • web-javascript • Google Chrome 81.0.4044.113
If you would like your app to run on android or web, consider running `flutter create .` to generate projects for these platforms.

$ flutter logs -d Leno
Showing Lenovo TB 8504F logs:
```

Since the android tablet showed logs when directly selected, it's clearly supported...

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Flutter logs should not attempt to filter the device list based on the current project, because it does not require a current project. Also fix disabled polling test

Fixes flutter/flutter#47996
Fixes https://github.com/flutter/flutter/issues/63550

### CODE REVIEW

File: packages/flutter_tools/lib/src/device.dart [hunk 1]
```diff
@@ -428,7 +435,7 @@ abstract class PollingDeviceDiscovery extends DeviceDiscovery {
 
   Future<List<Device>> pollingGetDevices({ Duration timeout });
 
-  Future<void> startPolling() async {
+  void startPolling() {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/device.dart:438]: Most of these methods do not need to be async - this made it much easier to fix the disabled test

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 2]
```diff
@@ -66,7 +66,7 @@ class DaemonCommand extends FlutterCommand {
 
 typedef DispatchCommand = void Function(Map<String, dynamic> command);
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/daemon.dart:67]: These changes just make the async APIs sync like the underlying code

**Comment by jonahwilliams** [comment]
@Hixie as the primary user of `flutter logs` would you like to review? 😃

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 3]
```diff
@@ -66,7 +66,7 @@ class DaemonCommand extends FlutterCommand {
...
 
-typedef CommandHandler = Future<dynamic> Function(Map<String, dynamic> args);
+typedef CommandHandler = FutureOr<dynamic> Function(Map<String, dynamic> args);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/daemon.dart:69]: FWIW, I would strongly prefer we never use `FutureOr`. It leads to all kinds of issues later.

Comment (Hixie) [reply packages/flutter_tools/lib/src/commands/daemon.dart:69]: not quite sure i understand where this ends up mattering

**Review by Hixie — APPROVED** [review]
LGTM module the FutureOr. But I suspect that means something else major has to change...?

File: packages/flutter_tools/lib/src/commands/daemon.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/daemon.dart]: Removed, its not necessary

**Comment by jonahwilliams** [comment]
Fixed the FutureOr issue :) , NBD

**Comment by jonahwilliams** [comment]
Landing to kick the tree - the last red test passed but the build status is stuck

**Pull request merged**